playwright-tests/tests/profile.spec.js

```diff
@@ -0,0 +1,174 @@
+        name: "Elliot",
+      });
+      await expect(profileName).toBeVisible();
+    });
+
+    test("Should navigate to Edit Profile and save correct data", async ({
+      page,
+    }) => {
+      const editButton = page.getByRole("button", { name: "Edit Profile" });
+      await editButton.click();
+
+      // expect input field with label Name to be visible
+      const nameInput = page.getByLabel("Name");
+      await expect(nameInput).toBeVisible();
+
+      const avatarInput = await page.locator("input[type=file]");
+      await avatarInput.setInputFiles(
+        path.join(__dirname, "./assets/black.png"),
+      );
+
+      await page.route("**/add", async (route) => {
+        const modifiedResponse = {
+          status: 200,
+          contentType: "application/json",
+          body: JSON.stringify({ cid: "simple_cid" }),
+        };
+
+        // Fulfill the route with the modified response
+        await route.fulfill(modifiedResponse);
+      });
+
+      await page.getByPlaceholder("Enter full name").fill("Someone");
+
+      const description = await page
+        .frameLocator("iframe")
+        .locator('textarea[name="textarea"]');
+      await description.fill("Someone");
+
+      await page.getByPlaceholder("eg. United States").fill("Someone");
+
+      await page.getByPlaceholder("twitter handle").fill("Someone");
+      await page.getByPlaceholder("github handle").fill("Someone");
+      await page.getByPlaceholder("telegram handle").fill("Someone");
+      await page.getByPlaceholder("website link").fill("Someone.com");
+
+      const expectedResult = {
+        "anybody.near": {
+          profile: {
+            name: "Someone",
+            description: "Someone",
+            image: {
+              ipfs_cid: "simple_cid",
+            },
+            location: "Someone",
+            linktree: {
+              twitter: "Someone",
+              github: "Someone",
+              telegram: "Someone",
+              website: "Someone.com",
+            },
+          },
+        },
+      };
+      // click button that triggers transaction
+      await page.getByRole("button", { name: "Save Profile" }).nth(0).click();
+      const transactionObj = JSON.parse(
+        await page.locator("div.modal-body code").innerText(),
+      );
+      // do something with transactionObj
+      expect(transactionObj).toMatchObject(expectedResult);
+    });
+  });
+
+  test.describe("User not logged in", () => {
+    test.use({
+      storageState: "playwright-tests/storage-states/wallet-not-connected.json",
+    });
+
+    test("Should show login prompt if no accountId is passed", async ({
+      page,
+    }) => {
+      await page.goto(`/${ROOT_SRC}?page=profile`);
+      const loginPrompt = page.getByText(
+        "AccountID prop or signed in account is required",
+      );
+      await expect(loginPrompt).toBeVisible();
+    });
+
+    test("Should show profile page when accountId is passed", async ({
```

Comment:
```suggestion
    test("should show profile page when accountId is passed", async ({
```
Yeah this is repeated test